src/Microsoft.Health.Fhir.SqlServer/Features/Storage/TvpRowGeneration/Merge/MergeResourceWrapper.cs

```diff
@@ -0,0 +1,40 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.Fhir.Core.Features.Persistence;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Storage
+{
+    internal class MergeResourceWrapper
+    {
+        internal MergeResourceWrapper(ResourceWrapper resourceWrapper, long resourceSurrogateId, bool keepHistory, bool hasVersionToCompare)
+        {
+            ResourceWrapper = resourceWrapper;
```

Comment:
Should we add checks to avoid null values here?

Reply:
This is internal class.